src/utils.rs

```diff
@@ -45,32 +44,31 @@ pub fn is_text(mime_type: &str) -> bool {
 }
 
 /// Errors that can occur in `copy_data()`.
-#[derive(Fail, Debug)]
+#[derive(derive_more::Error, derive_more::Display, Debug)]
 pub enum CopyDataError {
-    #[fail(display = "Couldn't set the source file descriptor flags")]
-    SetSourceFdFlags(#[cause] nix::Error),
+    #[display(fmt = "Couldn't set the source file descriptor flags")]
+    SetSourceFdFlags(#[error(source)] nix::Error),
 
-    #[fail(display = "Couldn't set the target file descriptor flags")]
-    SetTargetFdFlags(#[cause] nix::Error),
+    #[display(fmt = "Couldn't set the target file descriptor flags")]
+    SetTargetFdFlags(#[error(source)] nix::Error),
 
-    #[fail(display = "Couldn't fork")]
-    Fork(#[cause] nix::Error),
+    #[display(fmt = "Couldn't fork")]
+    Fork(#[error(source)] nix::Error),
 
-    #[fail(display = "Couldn't close the source file descriptor")]
-    CloseSourceFd(#[cause] nix::Error),
+    #[display(fmt = "Couldn't close the source file descriptor")]
+    CloseSourceFd(#[error(source)] nix::Error),
 
-    #[fail(display = "Couldn't close the target file descriptor")]
-    CloseTargetFd(#[cause] nix::Error),
+    #[display(fmt = "Couldn't close the target file descriptor")]
+    CloseTargetFd(#[error(source)] nix::Error),
 
-    #[fail(display = "Couldn't wait for the child process")]
-    Wait(#[cause] nix::Error),
+    #[display(fmt = "Couldn't wait for the child process")]
+    Wait(#[error(source)] nix::Error),
 
-    #[fail(display = "Received an unexpected status when waiting for the child process: {:?}",
-           _0)]
-    WaitUnexpected(WaitStatus),
+    #[display(fmt = "Received an unexpected status when waiting for the child process: {:?}", _0)]
+    WaitUnexpected(#[error(ignore)] WaitStatus),
 
-    #[fail(display = "The child process exited with a non-zero error code: {}", _0)]
-    ChildError(i32),
+    #[display(fmt = "The child process exited with a non-zero error code: {}", _0)]
+    ChildError(#[error(ignore)] i32),
```

Comment:
Why do certain fields need an explicit ignore?

Reply:
I'm not sure. The derive macro complained that `i32` does not implement `Error` and I had to add this to get it compile.